packages/core-transaction-pool/lib/pool-wallet-manager.js

```diff
@@ -94,17 +94,18 @@ module.exports = class PoolWalletManager extends WalletManager {
     if (!recipient && recipientId) { // cold wallet
       recipient = new Wallet(recipientId)
       this.walletsByAddress[recipientId] = recipient
+    }
 
```

Comment:
Did you make sure that breaking apart this `if else` block won't have any side effects?

Reply:
sending data to cold wallet - so first if is true... so if the if-else remain together 

`  } else if (!sender.canApply(data)) {` is never executed and transaction is accepted with (0 fees for example or anything else.

-- 
we should then if - transaction is not applied/accepted - remove the cold wallet from wallet manager.. @fix

Reply:
mmm true that. Can you document this  in the code or maybe extract these lines as a function:
https://github.com/ArkEcosystem/core/blob/fc4a29e866f03cd26ebc0a0b8ca1e0206becc4ee/packages/core-database/lib/wallet-manager.js#L212-L218

replaced by `const recipient = this.findOrCreateWallet(recipientId)`

Reply:
@fix https://github.com/ArkEcosystem/core/pull/933/commits/3126c3713d0025873f0fef45ca7c26ddb866cbdc

Should be ready to be merged.